source/dotnet/wholesale-api/Edi.UnitTests/Validators/WholesaleServicesRequest/GridAreaValidatorTest.cs

```diff
@@ -0,0 +1,157 @@
+public class GridAreaValidatorTest
+{
+    private const string ValidGlnNumber = "qwertyuiopasd"; // Must be 13 characters to be a valid GLN
+    private static readonly ValidationError _missingGridAreaCode = new("Netområde er obligatorisk for rollen DDM / Grid area is mandatory for the role DDM.", "D64");
+    private static readonly ValidationError _invalidGridArea = new("Ugyldig netområde / Invalid gridarea", "E86");
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task Validate_WhenRequesterIsGridOwnerOfRequestedGridArea_ReturnsNoValidationErrors(
+        [Frozen] Mock<IGridAreaOwnerRepository> gridAreaOwnerRepository,
+        GridAreaValidationRule sut)
+    {
+        // Arrange
+        var gridAreaCode = "123";
+
+        gridAreaOwnerRepository.Setup(repo => repo.GetCurrentOwnerAsync(gridAreaCode, It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new GridAreaOwner(
+                Guid.NewGuid(),
+                gridAreaCode,
+                ValidGlnNumber,
+                Instant.MinValue,
+                0));
+
+        var message = new WholesaleServicesRequestBuilder()
+            .WithRequestedByActorId(ValidGlnNumber)
+            .WithRequestedByActorRole(DataHubNames.ActorRole.GridOperator)
+            .WithGridArea(gridAreaCode)
+            .Build();
+
+        // Act
+        var errors = await sut.ValidateAsync(message);
+
+        // Assert
+        errors.Should().BeEmpty();
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task Validate_WhenRequesterIsNotGridOwnerOfRequestedGridArea_ReturnsExpectedValidationError(
+        [Frozen] Mock<IGridAreaOwnerRepository> gridAreaOwnerRepository,
+        GridAreaValidationRule sut)
+    {
+        // Arrange
+        var gridAreaCode = "123";
+        var gridAreaOwner = "qwertyuiopash";
+
+        gridAreaOwnerRepository.Setup(repo => repo.GetCurrentOwnerAsync(gridAreaCode, It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new GridAreaOwner(
+                Guid.NewGuid(),
+                gridAreaCode,
+                gridAreaOwner,
+                Instant.MinValue,
+                0));
+
+        var message = new WholesaleServicesRequestBuilder()
+            .WithRequestedByActorId(ValidGlnNumber)
+            .WithRequestedByActorRole(DataHubNames.ActorRole.GridOperator)
+            .WithGridArea(gridAreaCode)
+            .Build();
+
+        // Act
+        var errors = await sut.ValidateAsync(message);
+
+        // Assert
+        errors.Should().ContainSingle();
+
+        using var assertionScope = new AssertionScope();
+        var error = errors.Single();
+        error.Message.Should().Be(_invalidGridArea.Message);
+        error.ErrorCode.Should().Be(_invalidGridArea.ErrorCode);
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task Validate_WhenGridAreaCodeIsEmpty_ReturnsExpectedValidationError(
```

Comment:
```suggestion
    public async Task Validate_WhenGridAreaIsMissing_ReturnsExpectedValidationError(
```
I personally dont like the "code", so would this test name make sence?